### PR TITLE
fix(trade): transaction fees

### LINKED ISSUE #14823: Trading - button All sometimes fill more than available balance
**Describe the bug**

Suite version: 24.10.1

For some reason sometimes when I click `All` it fills more than available balance

https://github.com/user-attachments/assets/290a84d4-fc70-4aba-85ed-f0c758f88e19

### LINKED ISSUE #14835: Trading - all button sometimes does not fill amount at all
**Describe the bug**

Suite version: 24.10.1

Sometimes the `All` button does not fill anything

https://github.com/user-attachments/assets/d30aa515-233d-493c-ab35-a4e813c1f77f

### PR DESCRIPTION
**Pull request by adderpositive**
## Description
- Fixed decimals places - every selected account/token should support their decimals places (e.g. Solana supports max. 9 decimal places and USDT on Solana network should support max. 6)
- Fixed inappropriate recalculating of fees and all sent cryptocurrency
- Fixed All button always update input

## Related Issue
Resolve https://github.com/trezor/trezor-suite/issues/14823
Resolve https://github.com/trezor/trezor-suite/issues/14835

### CODE REVIEW

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketAddressOptions.tsx [hunk 1]
```diff
@@ -114,9 +117,14 @@ export const CoinmarketAddressOptions = <TFieldValues extends CoinmarketBuyAddre
                     formatOptionLabel={(accountAddress: AccountAddress) => {
                         if (!accountAddress || !account || !receiveSymbol) return null;
 
-                        const networkDecimals = getNetworkDecimals(
-                            networks[account.symbol].decimals,
-                        );
+                        const sendCryptoSelect = isCoinmarketExchangeContext(context)
+                            ? context.getValues('sendCryptoSelect')
```
Comment (LukasRada) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketAddressOptions.tsx]: Extract the 'sendCryptoSelect' to constant or use the constant.

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketOfferExchange/CoinmarketOfferExchangeSendSwap.tsx [hunk 2]
```diff
@@ -70,14 +71,25 @@ const formatCryptoAmountAsAmount = (amount: number, baseAmount: number, decimals
...
+        sendTransaction,
+        getValues,
+    } = useCoinmarketFormContext<CoinmarketTradeExchangeType>();
     const { cryptoIdToCoinSymbol } = useCoinmarketInfo();
     const [slippage, setSlippage] = useState(selectedQuote?.swapSlippage ?? '1');
     const [customSlippage, setCustomSlippage] = useState(slippage);
     const [customSlippageError, setCustomSlippageError] = useState<
         (FieldError & { message: TranslationKey }) | undefined
     >();
+    const sendCryptoSelect = getValues('sendCryptoSelect');
```
Comment (LukasRada) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketOfferExchange/CoinmarketOfferExchangeSendSwap.tsx]: Extract the 'sendCryptoSelect' to constant or use the constant.

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketVerify/CoinmarketVerifyOptionsItem.tsx [hunk 3]
```diff
@@ -54,6 +57,11 @@ export const CoinmarketVerifyOptionsItem = ({
                             balance={formattedBalance}
                             cryptoSymbolLabel={symbol.toLocaleUpperCase()}
                             networkSymbol={symbol}
+                            sendCryptoSelect={
+                                isCoinmarketExchangeContext(context)
+                                    ? context.getValues('sendCryptoSelect')
```
Comment (LukasRada) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketVerify/CoinmarketVerifyOptionsItem.tsx]: Extract the 'sendCryptoSelect' to constant or use the constant.

File: packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketCurrencySwitcher.ts [hunk 4]
```diff
@@ -48,8 +48,12 @@ export const useCoinmarketCurrencySwitcher = <T extends CoinmarketAllFormProps>(
     const { setValue, getValues, control } =
         methods as unknown as UseFormReturn<CoinmarketAllFormProps>;
     const { shouldSendInSats } = useBitcoinAmountUnit(account.symbol);
-    const networkDecimals = getNetworkDecimals(network?.decimals);
     const cryptoInputValue = useWatch({ control, name: inputNames.cryptoInput });
+    const sendCryptoSelect = getValues('sendCryptoSelect');
```
Comment (LukasRada) [inline packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketCurrencySwitcher.ts]: Extract the 'sendCryptoSelect' to constant or use the constant.

**Review by LukasRada — CHANGES_REQUESTED** [review]

**Review by LukasRada — APPROVED** [review]

File: packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketFormActions.ts [hunk 5]
```diff
@@ -203,8 +177,7 @@ export const useCoinmarketFormActions = <T extends CoinmarketSellExchangeFormPro
             // set token address for ERC20 transaction to estimate the fees more precisely
             setValue(FORM_OUTPUT_ADDRESS, selected?.contractAddress ?? '');
         }
-
-        if (networkSymbol === 'sol') {
+        if (networkSymbol === 'sol' && !selected?.contractAddress) {
```
Comment (tomasklim) [inline packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketFormActions.ts]: nit: would be better to use `networkType` === `solana` (you have it from account

File: packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketFormActions.ts [hunk 6]
```diff
@@ -203,8 +177,7 @@ export const useCoinmarketFormActions = <T extends CoinmarketSellExchangeFormPro
             // set token address for ERC20 transaction to estimate the fees more precisely
```
Comment (tomasklim) [inline packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketFormActions.ts:176]: please again, use `networkType === 'ethereum'` 
btw this place smells like a bug ❗ you do it only for ethereum, not for polygon, bnb,...

**Review by tomasklim — APPROVED** [review]

**Pull request merged**